plugins/filter/filters.py

```diff
@@ -32,46 +33,67 @@ def filters(self):
             'resolve_principal': self.resolve_principal
         }
 
-    def normalize_sasl_protocol(self, protocol):
-        # Returns standardized value for sasl mechanism string
-        normalized = 'GSSAPI' if protocol.lower() == 'kerberos' \
-            else 'SCRAM-SHA-512' if protocol.upper() == 'SCRAM' \
-            else 'SCRAM-SHA-256' if protocol.lower() == 'scram256' \
-            else 'PLAIN' if protocol.upper() == 'PLAIN' \
-            else 'OAUTHBEARER' if protocol.upper() == 'OAUTH' \
-            else 'none'
-        return normalized
-
-    def kafka_protocol_normalized(self, sasl_protocol_normalized, ssl_enabled):
+    def normalize_sasl_protocol(self, protocols):
+        # Returns a list of standardized values for sasl mechanism strings
+        protocol_list = protocols.replace(" ", "").split(',')
```

Comment:
I think we should take an ansible list as input instead of space and comma separated string. Reason being we are parsing it to remove spaces. But in case someone isnt converting their tabs to spaces and they by mistake gave a tab in the inventory file then this replace wont work.

Reply:
We need to have backward compatibility as well, considering it as array input will break it.

Reply:
I think we should also be handling tabs here similar to spaces.

Reply:
I tested with tabs as well, that is also working.